src/mosquito/base.cr

```diff
@@ -20,7 +20,7 @@ module Mosquito
 
       TEXT
 
-      error += @@mapping.keys.map { |k| "- #{k}" }.join "\n"
+      error += @@mapping.map { |k,v| "#{k} => #{v}" }.join "\n"
       error += "\n\n"
```

Comment:
This is prob a good use case for `String.build` with the `io` overload of `#join`. Save yourself the extra array and some string allocations.

Something like:
```cr
error = String.build do |io|
  io << <<-TEXT
  Could not find a job class for type "#{type}", perhaps you forgot to register it?
  Current known types are:\n
  TEXT

  @@mapping.join io, '\n' { |(k, v), io| io << "#{k} => #{v}" }

  io << '\n' << '\n'
end
```

Reply:
Nice!

Reply:
I'm not sure I see the benefit of #join(io, ...) here, but the builder is a good refactor.

```crystal
      error = String.build do |s|
        s << <<-TEXT
                   Could not find a job class for type "#{type}", perhaps you forgot to register it?

                   Current known types are:

                   TEXT

        @@mapping.each { |k, v| s << "#{k}=>#{v}\n" }

        s << "\n\n"
      end

      raise KeyError.new(error)
```